backend/fms_core/services/profile.py

```diff
@@ -0,0 +1,48 @@
+from collections import defaultdict
+from fms_core.models import Profile, FreezemanUser
+from fms_core.schema_validators import PREFERENCES_VALIDATOR
+
+def get_default_profile() -> Profile:
+    return Profile.objects.get(name="Default")
+
+def get_profile(user_id: int) -> Profile:
+    from fms_core.services.freezeman_user import get_or_create_freezeman_user
+    fm_user = get_or_create_freezeman_user(user_id)
+    return fm_user.profile
+
+def update_preferences(user_id: int, new_preferences: dict) -> tuple[Profile, list[str], list[str]]:
+    from fms_core.services.freezeman_user import get_or_create_freezeman_user
+    fm_user = get_or_create_freezeman_user(user_id)
+
+    errors = []
+    warnings = []
+
+    for error in PREFERENCES_VALIDATOR.validator.iter_errors(new_preferences):
+        path = "".join(f'["{p}"]' for p in error.path)
+        errors.append(f"{path}: {error.message}" if error.path else error.message)
+    if errors:
+        return fm_user.profile, errors, warnings
+
+    parent_profile = fm_user.profile.parent if fm_user.profile.parent else fm_user.profile
+    # Remove any new preferences that are the same as the parent's preferences
+    # parent should have all preference settings
+    for k, v in parent_profile.preferences.items():
+        if new_preferences.get(k, None) == v:
+            del new_preferences[k]
+
+    if not new_preferences and fm_user.profile.parent is None:
+        # don't update user preferences if the user hasn't personalized yet anyway which means they stuck with default
+        return fm_user.profile, errors, warnings
+
+    if fm_user.profile.parent is None:
```

Comment:
where is the else ? What happens when a profile has a parent ? this is where the actual update_preferences happens.